docs/contributing/CONTRIBUTING.md

```diff
@@ -0,0 +1,131 @@
+
+#### 3. Set up a development environment.
+Refer to the repository README for instructions on how to set up a development environment so you can test any local code changes you make.
+Note that the steps to set up a development environment (usually under a "Local installation", "Manual installation", or "Development environment" section) are generally different than those used to install the tool purely as a user, which usually appears at the top of the README.
+
+For example, for the [Neurobagel CLI](https://github.com/neurobagel/bagel-cli):
+
+- ["Development" mode installation steps](https://github.com/neurobagel/bagel-cli?tab=readme-ov-file#installation)
+- [Normal (non-development mode) installation steps](https://github.com/neurobagel/bagel-cli?tab=readme-ov-file#development-environment)
+
+If you are working on a repository written in Python, we recommend that you also set up automatic code formatting and linting according to the project's code style, 
+which is pre-configured in the development environment for these repositories using [pre-commit](https://pre-commit.com/).
+
+Run the following from the repository root of your local clone:
+```bash
+pre-commit install
+```
+
+Now, a number of code linters and formatters will automatically run when you attempt to make a commit, which will keep your changes consistent with the rest of the codebase.
+
+#### 4. [Create a new branch](https://help.github.com/articles/creating-and-deleting-branches-within-your-repository/) to make the proposed changes
+Please consider using descriptive branch names. Some examples:
+
+- `<username>/<issue-identifier>` (`jsmith/fix-1234`)
+- `<username>/<brief-change-title>` (`jsmith/add-logging`, `jsmith/enh/add-logging`)
+
+Once you are satisfied with your local changes, [add, commit, and push](https://docs.github.com/en/repositories/working-with-files/managing-files/adding-a-file-to-a-repository#adding-a-file-to-a-repository-using-the-command-line) them to your branch on your forked repository on GitHub. 
+
+#### 5. [Open a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork)
+See the below sections for information on [how to submit your pull request](#pull-request-guidelines) and [what to expect in a pull request review](#pull-request-reviews).
+
+## Pull request guidelines
+
+When you first open a pull request, you should automatically see a template in the pull request body that looks something like [this](https://github.com/neurobagel/.github/blob/main/.github/pull_request_template.md?plain=1) that you can fill out.
+The template is designed to make it easier for maintainers to review your pull request, but feel free to add any additional information that you feel is useful or necessary.
```

Comment:
I think it's a good idea to mention "The checks in the template should remain intact as they're for the maintainer to go over during review."

Reply:
Wdyt about making this more visible in the PR template itself? I'm hoping to avoid having duplication of info where possible, in case we change how we use the PR template in the future.

As part of https://github.com/neurobagel/.github/issues/9, I'm thinking to add a non-commented note to our [checklist](https://github.com/neurobagel/.github/blob/main/.github/pull_request_template.md#checklist) reminding PR authors to leave the checkboxes for the PR reviewer.

Reply:
Good call.